kahuna/public/js/upload/dnd-uploader.js

```diff
@@ -21,9 +21,12 @@ dndUploader.controller('DndUploaderCtrl',
                             apiPoll, witnessApi) {
 
     var ctrl = this;
+    const gridThumbnailPattern = /https:\/\/media-service([0-9-a-z]+)thumbbucket([0-9-a-z]+)/;
```

Comment:
The only thing about this pattern is that it's coupled with CF configuration that lives outside this repo. To make this more generic the API would have to tell kahuna what the domain our S3 buckets live under. Sounds like quite a faff, so maybe this is OK, but could we have a TODO to highlight that this is a bit of a hack?